### ISSUE TITLE
L006 false positive, wants extra whitespace in case statement

### ISSUE THREAD

**Issue by pwildenhain** [alanmcruickshank/sqlfluff#272]
👋 Hey there -- just want to take a second to say thanks for dedicating so much to the project, we've fallen in love 😍 with it where I work (same place as @cnolanminich if you didn't already guess)

It seems like sqlfluff wants more whitespace _before_ the equal sign on line 3 (and adding the extra whitespace silences the lint).

test.sql
```sql
select
    case
        when cast(zip_code as integer) = 000000 then 'city1'
        else 'other city'
    end as zip_code_group
from my_table
```
Output
```
$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   3 | P:  40 | L006 | Operators should be preceded by a space.
```

I believe this is because of the function that precedes the `=` operator, because if I remove the `cast` then everything works as expected

test.sql
```
select
    case
        when zip_code = 000000 then 'city1'
        else 'other city'
    end as zip_code_group
from my_table
```

🤔 This reminds me somewhat of #207 if that's helpful

**Comment by pwildenhain** [comment]
🤦 ANNNDDD the reason it is similar is because I was working with a previous version -- carry on!
